src/index.ts

```diff
@@ -156,26 +160,26 @@ export default function addVariableDeclarations(
       }
     },
 
-    enter(path) {
+    enter(path: NodePath<t.Node>) {
       // ObjectMethod and ClassMethod nodes are strange in that their key name
       // is in the outer scope, not the method scope, so we get the wrong scope
       // if we use the usual scope enter and exit hooks. To work around this,
       // pretend to be one scope higher while in the key, then restore the state
       // afterward.
-      if ((path.parent.type === 'ObjectMethod' && path.key === 'key') ||
-          (path.parent.type === 'ClassMethod' && path.key === 'key')) {
-        savedStates.push(state);
-        state = state.parentState;
+      if ((t.isObjectMethod(path.parent) && path.key === 'key') ||
+          (t.isClassMethod(path.parent) && path.key === 'key')) {
+        savedStates.push(getState());
+        state = getState().parentState;
       }
     },
 
-    exit(path) {
-      if ((path.parent.type === 'ObjectMethod' && path.key === 'key') ||
-          (path.parent.type === 'ClassMethod' && path.key === 'key')) {
-        state = savedStates.pop();
+    exit(path: NodePath<t.Node>) {
+      if ((t.isObjectMethod(path.parent) && path.key === 'key') ||
+          (t.isClassMethod(path.parent) && path.key === 'key')) {
+        state = savedStates.pop() || null;
       }
     },
-  });
+  } as any);
```

Comment:
This is only here until https://github.com/DefinitelyTyped/DefinitelyTyped/pull/16823 lands.